hash_practice/exercises.py

```diff
@@ -1,29 +1,115 @@
 
+from attr import has
+
+
 def grouped_anagrams(strings):
-    """ This method will return an array of arrays.
+    """ 
+    - Use a hash table to solve a coding problem
+    - Identify how a hash table can produce a more attractive runtime over alternative solutions
+    Given an array of strings, group anagrams together.
+    ### Example
+    ```
+    Input: ["eat", "tea", "tan", "ate", "nat", "bat"],
+    Output:
+    [
+    ["ate","eat","tea"],
+    ["nat","tan"],
+    ["bat"]
+    ]
+    ```
+    Note:
+    - All inputs will be in lowercase.
+    - The order of your output does not matter
+
+        This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        
+        Time Complexity: O(m * nlogn) - | python sorting algorithm -> O(n. logn).
+        Space Complexity: O(n) 
     """
-    pass
+    # return empty list if no words 
+    if len(strings) == 0:
+        return []
+
+    hash_table = {}
+    for item in strings:
+        # sort time complexity => O(n. logn).
+        string = "".join(sorted(item))
+        if string not in hash_table:
+            hash_table[string] = [item]
+        else:
+            hash_table[string].append(item)
+
+    print(hash_table.values())
+    return list(hash_table.values())
+
+
+print(grouped_anagrams(["eat", "tae", "tea", "eta", "aet", "ate"]))

```

Comment:
In general, make sure to remove testing code like this before you submit your PRs.